### PR TITLE
Exclude non en-US users from download funnel experiment

### LINKED ISSUE #10686: Exclude non en-US users from download funnel experiment
As discussed at https://github.com/mozilla/bedrock/issues/10207#issuecomment-861355640, we need to limit the download funnel experiment to only en-US users, as anyone with a different language should not be directed to the new experimental download page.

We might be able to hack this into the current experiment code, but it might be better to add a feature to the framework that allows us to exclude users from an experiment based on language. This is something that has been discussed in https://github.com/mozilla/addons-frontend/issues/10598 and that we will want to have in place at some point, so now seems as good a time as any to do it.

I will open a separate issue specifically about allowing for configuring an experiment based on language.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10686

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@willdurand I see you self-requested review, thanks! I have rebased this and it's ready to be reviewed.

**Comment by willdurand** [comment]
> @willdurand I see you self-requested review, thanks! I have rebased this and it's ready to be reviewed.

@bobsilverberg I'll look into it soon, but there are test failures.

File: tests/unit/amo/experiments/test_20210531_download_funnel_experiment.js [hunk 1]
```diff
@@ -3,22 +3,40 @@ import { dispatchClientMetadata, userAgents } from 'tests/unit/helpers';
...
+  it.each(nonFirefoxBrowsers)('includes non-Firefox users: %s', (userAgent) => {
     const { state } = dispatchClientMetadata({ userAgent });
     expect(shouldExcludeUser({ state })).toEqual(false);
   });
+
+  it.each(nonFirefoxBrowsers)(
+    'excludes non-en-US users on non-Firefox browsers: %s',
+    (userAgent) => {
+      const { state } = dispatchClientMetadata({ lang: 'fr', userAgent });
+      expect(shouldExcludeUser(state)).toEqual(true);
```
Comment (willdurand) [inline tests/unit/amo/experiments/test_20210531_download_funnel_experiment.js:31]:
```suggestion
      expect(shouldExcludeUser({ state })).toEqual(true);
```

File: tests/unit/amo/experiments/test_20210531_download_funnel_experiment.js [hunk 2]
```diff
@@ -3,22 +3,40 @@ import { dispatchClientMetadata, userAgents } from 'tests/unit/helpers';
...
+    },
+  );
+
+  it.each(nonFirefoxBrowsers)(
+    'includes en-US users on non-Firefox browsers: %s',
+    (userAgent) => {
+      const { state } = dispatchClientMetadata({ lang: 'en-US', userAgent });
+      expect(shouldExcludeUser(state)).toEqual(false);
```
Comment (willdurand) [inline tests/unit/amo/experiments/test_20210531_download_funnel_experiment.js:39]:
```suggestion
      expect(shouldExcludeUser({ state })).toEqual(false);
```

**Review by willdurand — APPROVED** [review]
r+wc

**Pull request merged**